src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/Target_ARM/ARMEmitter.cs

```diff
@@ -32,12 +61,44 @@ public void EmitPUSH(Register reg)
             Builder.EmitShort((short)(0x0d04 + ((byte)reg << 12)));
         }
 
+        // pop reg
+        public void EmitPOP(Register reg)
+        {
+            Builder.EmitByte(0x5d);
+            Builder.EmitByte(0xf8);
+            Builder.EmitShort((short)(0x0b04 + ((byte)reg << 12)));
+        }
+
         // mov reg, reg
+        // reg range: 0-7
```

Comment:
The encoding used by this method uses 4 bits for registers. E.g. the encoding for `mov r8, r9` is 0xC8 0x46, and this method will produce it fine.